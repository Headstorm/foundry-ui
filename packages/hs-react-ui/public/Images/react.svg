<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 19.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="561.8px" height="499.9px" viewBox="-80.4 151.2 561.8 499.9" style="enable-background:new -80.4 151.2 561.8 499.9;"
	 xml:space="preserve">
<style type="text/css">
	.st0{fill:#939598;}
</style>
<circle class="st0" cx="200.5" cy="400.6" r="50.2"/>
<path class="st0" d="M200.5,515.6c-70.5,0-132.1-8.3-178.2-24.1c-29.9-10.2-55.3-23.8-73.4-39.3c-19.2-16.4-29.4-34.3-29.4-51.6
	c0-33.2,36.4-65.7,97.5-86.9c50-17.4,115.2-27.1,183.4-27.1c67,0,131.3,9.4,181,26.6c29.1,10,53.6,23,71,37.4
	c18.9,15.8,28.9,33.1,28.9,50c0,34.5-40.7,69.4-106.3,91.1C328.6,507.1,266.6,515.6,200.5,515.6z M200.5,310.6
	c-64.7,0-128.7,9.4-175.5,25.7c-40.5,14.1-81.4,39.5-81.4,64.3c0,10,7.4,21.8,21,33.4c15.8,13.5,38.4,25.6,65.5,34.8
	c43.6,14.9,102.6,22.8,170.4,22.8c63.6,0,122.9-8,167-22.7c61.7-20.5,89.9-49.8,89.9-68.3c0-9.5-7.2-20.7-20.3-31.6
	c-15.1-12.6-37.1-24.1-63.4-33.2C326.4,319.6,264.9,310.6,200.5,310.6z"/>
<path class="st0" d="M86.6,650.8c-10.2,0-19.2-2.2-26.8-6.6C31,627.6,21.1,579.8,33.1,516.3c9.9-52.1,34.1-113.3,68.2-172.4h0
	c33.5-58,73.7-109,113.4-143.5c23.2-20.2,46.7-35,67.9-42.8c23.1-8.5,43.1-8.5,57.7-0.1c29.9,17.2,39.8,70,25.8,137.6
	c-9.9,48-33.5,105.9-66.5,163.2c-35.2,61-73.2,110.2-109.9,142.3c-23.8,20.8-48.3,36-70.7,43.9C107.4,648.7,96.4,650.8,86.6,650.8z
	 M122.1,355.9c-32.3,56-56.2,116.1-65.4,164.9c-8,42.1-6.5,90.2,15,102.6c16.1,9.3,55,0.5,102.3-40.9c34.7-30.3,71-77.5,104.9-136.2
	c31.8-55.1,54.4-110.5,63.8-156c6-29.1,7.3-55.5,3.8-76.4c-3-17.9-9.4-30.6-18.2-35.6c-8.2-4.7-21.5-4.1-37.5,1.8
	c-18.5,6.8-39.4,20.1-60.4,38.4C192.8,251.3,154.3,300.1,122.1,355.9L122.1,355.9z"/>
<path class="st0" d="M314.4,651.1c-27.2,0-61.7-16.4-97.7-47.4c-40.2-34.6-81.1-86.1-115.3-145.2v0c-33.6-58-57.6-118.3-67.7-170
	c-5.9-30.2-7-57.9-3.2-80.2c4.2-24.3,14.1-41.6,28.8-50.1c29.8-17.3,80.5,0.5,132.1,46.4c36.6,32.5,75,81.9,108.1,139.1
	c35.3,61,59,118.5,68.4,166.3c6.1,31,7.1,59.8,2.8,83.2c-4.6,24.9-15,42.6-30,51.3C333.2,648.9,324.3,651.1,314.4,651.1z
	 M122.2,446.5c32.4,56,72.6,106.7,110.2,139c32.5,28,74.9,50.6,96.4,38.2c16.1-9.3,27.9-47.4,15.7-109
	c-9-45.2-31.7-100.2-65.6-158.9c-31.9-55.1-68.6-102.4-103.3-133.2c-22.2-19.7-44.4-34.1-64.2-41.5c-17-6.4-31.2-7.1-39.9-2.1
	c-8.2,4.7-14.3,16.6-17.2,33.4c-3.3,19.4-2.3,44.2,3.1,71.5C66.9,333,89.9,390.7,122.2,446.5L122.2,446.5z"/>
</svg>
