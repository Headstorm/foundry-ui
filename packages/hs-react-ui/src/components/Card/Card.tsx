import React, { ReactNode, MouseEvent } from 'react';
import styled, { StyledComponentBase } from 'styled-components';
import { darken } from 'polished';

import timings from '../../enums/timings';
import { Div } from '../../htmlElements';
import { SubcomponentPropsType } from '../commonTypes';
import { useTheme } from '../../context';
import { getShadowStyle } from '../../utils/styles';
import InteractionFeedback, {
  InteractionFeedbackProps,
} from '../InteractionFeedback/InteractionFeedback';
import FeedbackTypes from '../../enums/feedbackTypes';

const defaultOnClick = () => {};

export type CardContainerProps = {
  elevation: number;
  feedbackType: FeedbackTypes;
  onClick: (evt: MouseEvent) => void;
};

export const CardContainer = styled(Div)`
  ${({ elevation, feedbackType, onClick }: CardContainerProps) => {
    const { colors } = useTheme();

    return `
      ${onClick !== defaultOnClick ? 'cursor: pointer;' : ''}
      display: inline-flex;
      flex-flow: column nowrap;
      font-size: 1rem;
      border-radius: 0.25rem;
      border: ${!elevation ? `1px solid ${colors.grayXlight}` : '0px solid transparent'};
      transition:
        filter ${timings.slow},
        box-shadow ${timings.slow},
        border ${timings.normal},
        background-color ${timings.normal};
      ${getShadowStyle(elevation, colors.shadow)}
      background-color: ${colors.background};
      ${
        feedbackType === FeedbackTypes.simple && onClick !== defaultOnClick
          ? `
            &:active {
              background-color: ${
                colors.background !== 'transparent'
                  ? darken(0.1, colors.background)
                  : 'rgba(0, 0, 0, 0.1)'
              };
            }
          `
          : ''
      }
  `;
  }}
`;

export const Header = styled(Div)`
  ${({ hasBody, hasFooter }) => {
    const { colors } = useTheme();

    return `
      padding: 1.5rem 1.5rem ${hasBody || hasFooter ? '0rem' : ''};
      border-radius: 0.25rem 0.25rem 0rem 0rem;
      font-weight: bold;
      color: ${colors.grayDark};
    `;
  }}
`;

export const NoPaddingHeader = styled(Header)`
  padding: 0;
  overflow: hidden;
`;

export const Body = styled(Div)`
  ${() => {
    const { colors } = useTheme();

    return `
      padding: 1.5rem 1.5rem;
      color: ${colors.grayMedium};
    `;
  }}
`;

export const Footer = styled(Div)`
  ${() => {
    const { colors } = useTheme();

    return `
      padding: 1rem 1.5rem;
      display: flex;
      flex-flow: row wrap;

      justify-content: flex-end;

      color: ${colors.grayLight};

      border-radius: 0rem 0rem 0.25rem 0.25rem;
    `;
  }}
`;

const StyledFeedbackContainer = styled(InteractionFeedback.Container)`
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
`;
export interface CardProps {
  StyledContainer?: string & StyledComponentBase<any, {}>;
  StyledHeader?: string & StyledComponentBase<any, {}>;
  StyledBody?: string & StyledComponentBase<any, {}>;
  StyledFooter?: string & StyledComponentBase<any, {}>;

  containerProps?: SubcomponentPropsType;
  headerProps?: SubcomponentPropsType;
  bodyProps?: SubcomponentPropsType;
  footerProps?: SubcomponentPropsType;
  interactionFeedbackProps?: Omit<InteractionFeedbackProps, 'children'>;

  onClick?: (evt: MouseEvent) => void;

  header?: ReactNode;
  children?: ReactNode;
  footer?: ReactNode;

  elevation?: number;
  disableFeedback?: boolean;
  feedbackType?: FeedbackTypes;

  containerRef?: React.RefObject<HTMLDivElement>;
  headerRef?: React.RefObject<HTMLDivElement>;
  bodyRef?: React.RefObject<HTMLDivElement>;
  footerRef?: React.RefObject<HTMLDivElement>;
  interactiveFeedbackRef?: React.RefObject<HTMLDivElement>;
}

const Card = ({
  StyledContainer = CardContainer,
  StyledHeader = Header,
  StyledBody = Body,
  StyledFooter = Footer,

  containerProps,
  headerProps,
  bodyProps,
  footerProps,
  interactionFeedbackProps,

  containerRef,
  headerRef,
  bodyRef,
  footerRef,
  interactiveFeedbackRef,

  onClick = defaultOnClick,

  header,
  children,
  footer,

  elevation = 1,
  feedbackType = FeedbackTypes.ripple,
}: CardProps): JSX.Element | null => {
  const transitionProps = {
    ...InteractionFeedback.defaultTransitionProps,
    enter: {
      ...InteractionFeedback.defaultTransitionProps,
      r: 300,
    },
  };
  const hasHeader = Boolean(header);
  const hasBody = Boolean(children);
  const hasFooter = Boolean(footer);

  return (
    <StyledContainer
      onClick={onClick}
      elevation={elevation}
      feedbackType={feedbackType}
      {...containerProps}
      ref={containerRef}
    >
      {header && (
        <StyledHeader hasBody={hasBody} hasFooter={hasFooter} ref={headerRef} {...headerProps}>
          {header}
        </StyledHeader>
      )}
      {children && (
        <StyledBody hasHeader={hasHeader} hasFooter={hasFooter} ref={bodyRef} {...bodyProps}>
          {children}
        </StyledBody>
      )}
      {footer && (
        <StyledFooter hasHeader={hasHeader} hasBody={hasBody} ref={footerRef} {...footerProps}>
          {footer}
        </StyledFooter>
      )}
      {feedbackType !== FeedbackTypes.simple && onClick !== defaultOnClick && (
        <InteractionFeedback
          color="rgba(0,0,0,0.1)"
          transitionProps={transitionProps}
          StyledContainer={StyledFeedbackContainer}
          containerRef={interactiveFeedbackRef}
          {...interactionFeedbackProps}
        />
      )}
    </StyledContainer>
  );
};

Card.Header = Header;
Card.NoPaddingHeader = NoPaddingHeader;
Card.Footer = Footer;
Card.Body = Body;
Card.Container = CardContainer;

export default Card;
