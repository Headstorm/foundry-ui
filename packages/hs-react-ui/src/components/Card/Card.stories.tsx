import React from 'react';
import styled from 'styled-components';
import { text, number, boolean, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import { storiesOf } from '@storybook/react';

import Card, { Header, Footer } from './Card';
import colors from '../../enums/colors';
import timings from '../../enums/timings';
import fonts from '../../enums/fonts';
import feedbackTypes from '../../enums/feedbackTypes';

const design = {
  type: 'figma',
  url: 'https://www.figma.com/file/3r2G00brulOwr9j7F6JF59/Generic-UI-Style?node-id=102%3A0',
};

storiesOf('Card', module)
  .addParameters({ component: Card })
  .add(
    'Default',
    () => {
      return (
        <Card
          header={text('header', 'Card title')}
          footer={text('footer', 'Actionable buttons, whatever other stuff you want to pass in!')}
          elevation={number('elevation', 2, { range: true, min: -5, max: 5, step: 1 })}
          onClick={boolean('onClick', true) ? action('onClick') : undefined}
          feedbackType={select('feedbackType', feedbackTypes, feedbackTypes.ripple)}
        >
          {text(
            'children',
            'A Hello, World! program generally is a computer program that outputs or displays the message Hello, World!.',
          )}
        </Card>
      );
    },
    { design, centered: true },
  )
  .add('Themed', () => {
    const themeColors = {
      ...colors,
      background: 'beige',
      primary: 'purple',
    };

    const themeTimings = {
      ...timings,
      xSlow: '2s',
    };

    const ThemedContainer = styled.div`
      ${({ elevation = 0 }: { elevation: number }) => `
        border-radius: 1rem;
        width: fit-content;
        background-color: ${themeColors.background};

        transition: transform ${themeTimings.xSlow};
        transform: scale(${elevation * 0.05 + 1});

        ${fonts.body}
        font-size: 1rem;
        border: 1px solid ${themeColors.primary};
      `}
    `;

    const ThemedHeader = styled(Header)`
      font-family: Parchment, serif;
      line-height: 0;
      font-size: 4rem;
      padding-top: 2.5rem;
      padding-left: 0.75rem;
      padding-bottom: 1rem;
      text-transform: unset;
      color: ${themeColors.primary};
    `;

    const ThemedFooter = styled(Footer)`
      border-top: 1px solid ${themeColors.primary};
    `;

    return (
      <Card
        StyledContainer={ThemedContainer}
        StyledHeader={ThemedHeader}
        StyledFooter={ThemedFooter}
        header={text('header', 'Card title')}
        footer={text('footer', 'Actionable buttons, whatever other stuff you want to pass in!')}
        elevation={number('elevation', 0, { range: true, min: -5, max: 5, step: 1 })}
        onClick={action('onClick')}
        feedbackType={select('feedbackType', feedbackTypes, feedbackTypes.ripple)}
      >
        {text(
          'children',
          'A Hello, World! program generally is a computer program that outputs or displays the message Hello, World!.',
        )}
      </Card>
    );
  })
  .add(
    'Ref',
    () => {
      const cardContainerRef = React.createRef<HTMLDivElement>();
      const cardHeaderRef = React.createRef<HTMLDivElement>();
      const cardBodyRef = React.createRef<HTMLDivElement>();
      const cardFooterRef = React.createRef<HTMLDivElement>();
      const interactiveFeedbackRef = React.createRef<HTMLDivElement>();
      const onClick = (e: React.SyntheticEvent) => {
        e.preventDefault();
        action('onClick')(
          `container width x height: ${cardContainerRef.current?.clientWidth} x ${cardContainerRef.current?.clientHeight}
          header width x height: ${cardHeaderRef.current?.clientWidth} x ${cardHeaderRef.current?.clientHeight}
          body width x height: ${cardBodyRef.current?.clientWidth} x ${cardBodyRef.current?.clientHeight}
          footer width x height: ${cardFooterRef.current?.clientWidth} x ${cardFooterRef.current?.clientHeight}
          interactive width x height: ${interactiveFeedbackRef.current?.clientWidth} x ${interactiveFeedbackRef.current?.clientHeight}`,
        );
      };
      return (
        <Card
          header={text('header', 'View the Actions tab below')}
          footer={text(
            'footer',
            'Try adjusting the width of the viewport. New clicks will return the updated dimensions for each element.',
          )}
          elevation={number('elevation', 2, { range: true, min: -5, max: 5, step: 1 })}
          onClick={boolean('onClick', true) ? e => onClick(e) : undefined}
          feedbackType={select('feedbackType', feedbackTypes, feedbackTypes.ripple)}
          containerRef={cardContainerRef}
          headerRef={cardHeaderRef}
          bodyRef={cardBodyRef}
          footerRef={cardFooterRef}
          interactiveFeedbackRef={interactiveFeedbackRef}
        >
          {text(
            'children',
            'Then click anywhere on the Card to see the width/height of the child elements calculated via the Ref props!',
          )}
        </Card>
      );
    },
    { design, centered: true },
  );
