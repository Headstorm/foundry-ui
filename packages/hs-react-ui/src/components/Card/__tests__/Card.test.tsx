import React from 'react';
import styled from 'styled-components';
import { render, waitFor, configure } from '@testing-library/react';
import Card, { Body } from '../Card';
import FeedbackTypes from '../../../enums/feedbackTypes';
import { axe, toHaveNoViolations } from 'jest-axe';
expect.extend(toHaveNoViolations);

configure({ testIdAttribute: 'data-test-id' });

const testId = 'foundry-card';

describe('Card', () => {
  describe('Accessibility Tests', () => {
    it('Should pass accessibility test with default props', async () => {
      const component = <Card onClick={() => {}} aria-label="aria-label-test" />;
      const { container } = render(component);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });

  it('shows Card with default props', async () => {
    const { container, getByTestId } = render(
      <Card onClick={() => {}} containerProps={{ 'data-test-id': testId }} />,
    );
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Card with non-default elevation', async () => {
    const { container, getByTestId } = render(
      <Card onClick={() => {}} containerProps={{ 'data-test-id': testId }} elevation={3} />,
    );
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Card with default feedback', async () => {
    const { container, getByTestId } = render(
      <Card onClick={() => {}} containerProps={{ 'data-test-id': testId }} />,
    );
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Card with simple feedback and no onClick', async () => {
    const { container, getByTestId } = render(
      <Card containerProps={{ 'data-test-id': testId }} feedbackType={FeedbackTypes.simple} />,
    );
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Card with ripple feedback and no onClick', async () => {
    const { container, getByTestId } = render(
      <Card containerProps={{ 'data-test-id': testId }} feedbackType={FeedbackTypes.ripple} />,
    );
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Card with simple feedback with onClick', async () => {
    const { container, getByTestId } = render(
      <Card
        onClick={() => null}
        containerProps={{ 'data-test-id': testId }}
        feedbackType={FeedbackTypes.simple}
      />,
    );
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('shows Card with ripple feedback with onClick', async () => {
    const { container, getByTestId } = render(
      <Card
        onClick={() => null}
        containerProps={{ 'data-test-id': testId }}
        feedbackType={FeedbackTypes.ripple}
      />,
    );
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });
});
