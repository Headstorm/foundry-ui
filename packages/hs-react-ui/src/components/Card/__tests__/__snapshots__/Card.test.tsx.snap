// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card header bottom padding should be 0rem when body is present 1`] = `
.c3 {
  position: relative;
}

.c5 {
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
}

.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  background-color: #fff;
}

.c1 {
  padding: 1.5rem 1.5rem 0rem;
  border-radius: 0.25rem 0.25rem 0rem 0rem;
  font-weight: bold;
  color: #252c35;
}

.c2 {
  padding: 1.5rem 1.5rem;
  color: #51575d;
}

.c4 {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="1"
  >
    <div
      class=" c1"
    >
      Header
    </div>
    <div
      class=" c2"
    >
      Body
    </div>
    <div
      class="c3 c4"
    >
      <svg
        class="c5"
        height="0px"
        viewBox="0 0 0 0"
        width="0px"
      />
    </div>
  </div>
</div>
`;

exports[`Card header bottom padding should be 0rem when footer is present 1`] = `
.c3 {
  position: relative;
}

.c5 {
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
}

.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  background-color: #fff;
}

.c1 {
  padding: 1.5rem 1.5rem 0rem;
  border-radius: 0.25rem 0.25rem 0rem 0rem;
  font-weight: bold;
  color: #252c35;
}

.c2 {
  padding: 1rem 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  color: #7c8186;
  border-radius: 0rem 0rem 0.25rem 0.25rem;
}

.c4 {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="1"
  >
    <div
      class=" c1"
    >
      Header
    </div>
    <div
      class=" c2"
    >
      Footer
    </div>
    <div
      class="c3 c4"
    >
      <svg
        class="c5"
        height="0px"
        viewBox="0 0 0 0"
        width="0px"
      />
    </div>
  </div>
</div>
`;

exports[`Card header bottom padding should be 1.5rem without body and footer 1`] = `
.c2 {
  position: relative;
}

.c4 {
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
}

.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  background-color: #fff;
}

.c1 {
  padding: 1.5rem 1.5rem;
  border-radius: 0.25rem 0.25rem 0rem 0rem;
  font-weight: bold;
  color: #252c35;
}

.c3 {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="1"
  >
    <div
      class=" c1"
    >
      Header
    </div>
    <div
      class="c2 c3"
    >
      <svg
        class="c4"
        height="0px"
        viewBox="0 0 0 0"
        width="0px"
      />
    </div>
  </div>
</div>
`;

exports[`Card shows Card with default feedback 1`] = `
.c1 {
  position: relative;
}

.c3 {
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
}

.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  background-color: #fff;
}

.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="1"
  >
    <div
      class="c1 c2"
    >
      <svg
        class="c3"
        height="0px"
        viewBox="0 0 0 0"
        width="0px"
      />
    </div>
  </div>
</div>
`;

exports[`Card shows Card with default props 1`] = `
.c1 {
  position: relative;
}

.c3 {
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
}

.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  background-color: #fff;
}

.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="1"
  >
    <div
      class="c1 c2"
    >
      <svg
        class="c3"
        height="0px"
        viewBox="0 0 0 0"
        width="0px"
      />
    </div>
  </div>
</div>
`;

exports[`Card shows Card with non-default elevation 1`] = `
.c1 {
  position: relative;
}

.c3 {
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
}

.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.07538939058881129rem 0.75rem rgba(39,47,78,0.3741399931653331));
  filter: drop-shadow(0rem 0.07538939058881129rem 0.75rem rgba(39,47,78,0.3741399931653331));
  background-color: #fff;
}

.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="3"
  >
    <div
      class="c1 c2"
    >
      <svg
        class="c3"
        height="0px"
        viewBox="0 0 0 0"
        width="0px"
      />
    </div>
  </div>
</div>
`;

exports[`Card shows Card with ripple feedback and no onClick 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  background-color: #fff;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="1"
  />
</div>
`;

exports[`Card shows Card with ripple feedback with onClick 1`] = `
.c1 {
  position: relative;
}

.c3 {
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
}

.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  background-color: #fff;
}

.c2 {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="1"
  >
    <div
      class="c1 c2"
    >
      <svg
        class="c3"
        height="0px"
        viewBox="0 0 0 0"
        width="0px"
      />
    </div>
  </div>
</div>
`;

exports[`Card shows Card with simple feedback and no onClick 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  background-color: #fff;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="1"
  />
</div>
`;

exports[`Card shows Card with simple feedback with onClick 1`] = `
.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0px solid transparent;
  -webkit-transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  transition: filter .5s, box-shadow .5s, border .3s, background-color .3s;
  -webkit-filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  filter: drop-shadow(0rem 0.06653090368236622rem 0.25rem rgba(39,47,78,0.4292196217912265));
  background-color: #fff;
}

.c0:active {
  background-color: #e6e6e6;
}

<div>
  <div
    class=" c0"
    data-test-id="foundry-card"
    elevation="1"
  />
</div>
`;
